### PR TITLE
toposort: Make argument types less restrictive

### PR DESCRIPTION
**Pull request by Akuli**

### CODE REVIEW

File: stubs/toposort/toposort.pyi [hunk 1]
```diff
@@ -6,5 +7,5 @@ class CircularDependencyError(ValueError):
     data: dict[Any, set[Any]]
     def __init__(self, data: dict[Any, set[Any]]) -> None: ...
 
-def toposort(data: dict[_T, set[_T]]) -> Iterator[set[_T]]: ...
-def toposort_flatten(data: dict[_T, set[_T]], sort: bool = ...) -> list[_T]: ...
+def toposort(data: SupportsItems[_T, set[_T]]) -> Iterator[set[_T]]: ...
```
Comment (srittau) [inline stubs/toposort/toposort.pyi:10]: This also calls `len()` on `data`.

**Review by srittau — CHANGES_REQUESTED** [review]

**Review by srittau — APPROVED** [review]

**Comment by Akuli** [comment]
This doesn't actually work, for some reason:

```
$ MYPYPATH=$HOME/typeshed/stubs/toposort/ mypy porcupine
porcupine/pluginloader.py:268: error: Argument 1 to "toposort" has incompatible type "Dict[PluginInfo, Set[PluginInfo]]"; expected "_SupportsItemsAndLen[PluginInfo, Iterable[PluginInfo]]"  [arg-type]
Found 1 error in 1 file (checked 69 source files)
```

**Comment by Akuli** [comment]
this is weird:

```python3
from typing import Generic, TypeVar
from _typeshed import SupportsItems

_KT_co = TypeVar("_KT_co", covariant=True)
_VT_co = TypeVar("_VT_co", covariant=True)

class SupportsItemsCustom(SupportsItems[_KT_co, _VT_co], Generic[_KT_co, _VT_co]):
    pass

x: dict[int, str] = {}
foo: SupportsItems[int, str] = x  # no error
bar: SupportsItemsCustom[int, str] = x  # error: Incompatible types in assignment
```

**Comment by JelleZijlstra** [comment]
You have to inherit directly from `Protocol` to make it a Protocol. Right now you defined a concrete class.

**Comment by Akuli** [comment]
Now it works.

**Review by srittau — APPROVED** [review]

**Pull request merged**